### PR TITLE
Update from xyz

### PR DESCRIPTION
**Pull request by lukasturcani**
Added a function to update the structure of molecules from xyz files and a missing test for updating the structure from .mol files.

### CODE REVIEW

File: stk/molecular/molecules/molecule.py [hunk 1]
```diff
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
...
+            conformer = self.mol.GetConformer(conformer).GetId()
+
+        with open(path, 'r') as f:
+            atom_count, _, *content = f.readlines()
+
+        # Check the atom count is correct.
+        num_atoms = self.mol.GetNumAtoms()
+        if int(atom_count) != num_atoms:
+            raise RuntimeError(
+                'The number of atoms in the xyz file does '
```
Comment (andrewtarzia) [inline stk/molecular/molecules/molecule.py]: output the number of atoms in the XYZ and number of atoms in the molecule

File: stk/molecular/molecules/molecule.py [hunk 2]
```diff
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
...
+                'not match the number of atoms in the molecule.'
+            )
+
+        # Save all the coords in the file.
+        new_coords = []
+        for i, line in enumerate(content):
+            element, *coords = line.split()
+            if element.isnumeric():
+                element = periodic_table[int(element)]
+
```
Comment (andrewtarzia) [inline stk/molecular/molecules/molecule.py]: should raise an error if this statement is False

File: stk/molecular/molecules/molecule.py [hunk 3]
```diff
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
...
+            raise RuntimeError(
+                'The number of atoms in the xyz file does '
+                'not match the number of atoms in the molecule.'
+            )
+
+        # Save all the coords in the file.
+        new_coords = []
+        for i, line in enumerate(content):
+            element, *coords = line.split()
+            if element.isnumeric():
```
Comment (andrewtarzia) [inline stk/molecular/molecules/molecule.py]: should raise an error if statement is false

File: stk/molecular/molecules/molecule.py [hunk 4]
```diff
@@ -1050,6 +1051,72 @@ def update_from_mol(self, path, conformer=-1):
...
+                raise RuntimeError(
+                    f'Atom {i} element does not match file.'
+                )
+
+            new_coords.append([float(i) for i in coords])
+
+        # Check that the correct number of atom
+        # lines was present in the file.
+        if i+1 != num_atoms:
+            raise RuntimeError(
```
Comment (andrewtarzia) [inline stk/molecular/molecules/molecule.py]: include number of atoms in file and molecule again

File: stk/molecular/molecules/molecule.py (hunk 3 above)
Comment (lukasturcani) [reply stk/molecular/molecules/molecule.py]: This if statement just converts the atomic number into the element symbol, it then compares the element symbol. If this is false, it means that the element symbol was there already so no conversion needs to be done. Of course, if the element symbol is not there but some garbage is, the code necessary to raise an error is already present.

**Comment by andrewtarzia** [comment]
All tests passed with latest version of pywindowx

**Pull request merged**